Assets/Scripts/PlayerAvatar.cs

```diff
@@ -90,6 +97,19 @@ public class PlayerAvatar : MonoBehaviour {
         transform.Translate(rb.velocity * Time.deltaTime);
     }
 
+    protected void handleAttack() {
+        fireCooldown = fireCooldown - Time.deltaTime;
+
+        Vector3 projectilePos = transform.position;
+        if (Input.GetButton("Fire1") && fireCooldown <= 0)
+        {
+            // Don't spawn right on top of player
+            projectilePos += new Vector3(1.0f, 0.0f, 0.0f);
+            newProjectile = Instantiate(projectilePrefab, projectilePos, Quaternion.identity) as GameObject;
+            fireCooldown = fireDelay;
+        }
+    }

```

Comment:
Here's how I'm handling input so far and then passing it down. https://github.com/CleverGameDev/Cleverfall/blob/1e828d41e35de7cf272f3500403b416e62391754/Assets/Scripts/HumanPlayer.cs#L113-L134

I think we should have this event handled in a consistent way. The playerInput already has an "attack" input, so I think you can just add an `OnAttack` handler in the `HumanPlayer` class to receive that input event.

![image](https://user-images.githubusercontent.com/102242/83107905-de54bd00-a073-11ea-852d-93595c0b9370.png)